confidential/secrets_manager.py

```diff
@@ -90,9 +90,10 @@ def decrypt_string(self, value) -> str:
         decrypted_string = self.decrypt_secret_from_aws(value[7:])
 
         # Check if the payload is serialized JSON
+        # TypeError handles None case
         try:
             result = json.loads(decrypted_string)
-        except json.decoder.JSONDecodeError:
+        except (json.decoder.JSONDecodeError, TypeError):
```

Comment:
I don't think this fixes the problem. It will set `result = None`. 

This happens in `decrypt_secret_from_aws`. If you don't have permission to decrypt, the response from `boto3` still contains a payload but it doesn't contain the secret. 

I think the correct thing to do is check for a bad response, and raise an appropriate error.

Reply:
Agree with @vagelim

Reply:
Cool, that was the other way solution I had. making the change